### PR TITLE
Discord Bot /aita Slash Command with Praw

### PR DESCRIPTION
**Pull request by erakai**
Am I the asshole?

### CODE REVIEW

File: config.py [hunk 1]
```diff
@@ -1,6 +1,9 @@
 import os
 # Discord bot access token. Located in .env file in base directory.
 TOKEN = os.environ['TOKEN']
+# You get these from developer settings in reddit, ask kai to get added as developer
```
Comment (ericpretzel) [inline config.py:4]: add me as a developer

File: extension/aita.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
+"""
+Author: Kai (Not Eric (Not Lance (Not Russell)))
```
Comment (ericpretzel) [inline extension/aita.py:2]: Who is this

File: extension/aita.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        if (decision == "None"): return "Unable to locate a top comment with a valid decision. {text}"
+        if (guess == decision): return f"{guess} was **correct**! {text}"
+        return f"{guess} was **incorrect**! {text}"
+
+class AITA(commands.Cog):
+    def __init__(self, bot: commands.Bot):
+        self.bot = bot
+
+    @slash_command(guild_ids=[config.WIDMARK_CLAN_GUILD_ID],
+        description="Are they the Asshole?")
```
Comment (ericpretzel) [inline extension/aita.py]:
```suggestion
        description="Is OP the asshole?")
```

File: extension/aita.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    async def aita(self, ctx: discord.ApplicationContext):
+        try:
+            # holy fucking shi ti commands without this response the entire thing beraks cause if you dont respond in 3 seconds the slash commands falls through and because i have intractions on the butotns the error i got is "unkown interaction" which is incredibly uselessd adn tells you nothing so the only solution is to find a stackoverflow post about thie error in ajvascript with 2 upvotes and see somethign taht says that you need a response immedaitely so discord doesnt berak it
```
Comment (ericpretzel) [inline extension/aita.py]:
```suggestion
```

File: extension/aita.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+    @slash_command(guild_ids=[config.WIDMARK_CLAN_GUILD_ID],
+        description="Are they the Asshole?")
+    async def aita(self, ctx: discord.ApplicationContext):
+        try:
+            # holy fucking shi ti commands without this response the entire thing beraks cause if you dont respond in 3 seconds the slash commands falls through and because i have intractions on the butotns the error i got is "unkown interaction" which is incredibly uselessd adn tells you nothing so the only solution is to find a stackoverflow post about thie error in ajvascript with 2 upvotes and see somethign taht says that you need a response immedaitely so discord doesnt berak it 
+            await ctx.defer()
+            post = await get_aita()
+        except Exception as e:
+            return await ctx.respond('Unable to find an asshole. Try the mirror.', ephemeral=True)
```
Comment (ericpretzel) [inline extension/aita.py:54]: What mirror

File: util/reddit_scraper.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+USER_AGENT = 'Mozilla/5.0 (X11; CrOS x86_64 8172.45.0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.64 Safari/537.36'
+
+async def get_aita():
+    r = asyncpraw.Reddit(user_agent=USER_AGENT, 
+                    client_id=config.REDDIT_ID, 
+                    client_secret=config.REDDIT_SECRET)
+    aita = await r.subreddit('AmItheAsshole')
+    posts = [post async for post in aita.hot(limit=POST_LIMIT)]
+    post = posts[random.randint(0, POST_LIMIT)]
+    while (post.stickied): post = posts[random.randint(0, POST_LIMIT)]
```
Comment (ericpretzel) [inline util/reddit_scraper.py:16]: Potential infinite loop?

**Review by lancemathias — APPROVED** [review]
NTA - your pr, your rules

**Review by ericpretzel — CHANGES_REQUESTED** [review]

File: util/reddit_scraper.py (hunk 6 above)
Comment (erakai) [reply util/reddit_scraper.py:16]: You can only have 2 pinned threads at once so no.

**Review by ericpretzel — APPROVED** [review]
yta lgtmt nwd htsmfu

**Pull request merged**